### ISSUE TITLE
[weak supervision] Feedback on skweak tutorial

### ISSUE THREAD

**Issue by dvsrepo** [recognai/rubrix#942]
Hi @ruanchaves, I finally found a free slot to go through the tutorial. Here's a detailed review:

1. I'd try to streamline the preprocessing part. Maybe create a preprocessed dataset and move the preprocessing part to an appendix? The in the preprocessing section we just describe the data format.
2. As a minor comment for the tag_set we could use `dataset["train"].features["ner_tags"].feature.names`.
3. We strip BI tags, I would add a short sentence saying why we need to do this.
4.I'd try to log the conll train split (without tags) in Rubrix right at the start, so you can test some of the heuristics (for example the company ones, I guess that as conll is an English corpus we could remove `sa` and `ag`, and maybe even find some other suffix that work more or less well?). Also in terms of the workflow and support by Rubrix I believe that including this step adds some value, as you are able to verify your hypotheses by directly searching.
5.  Question: In the gazetteers part, if the conclusion after exploring is that only the wiki gazetteer is working, we should explain what to do, do we don't make use of the other gazetteers? 

The most fundamental issue that I see is: *using conll for evaluation and only defining lfs and aggregating on companies and orgs*. I believe this comes from the fact that we are reusing much from the original skweak tutorial which actually builds a NER model with those entities only. This is likely one reason the final models are underperforming. More specifically:

1. Our `hmm_model = CustomHMM("hmm", ["COMPANY", "OTHER_ORG"])` ignores `PER` and other `MISC` entities from the pre-trained model lf. I have checked this after the first Rubrix logging, you can see there are no `PER`, etc. In any case, if we don't add lfs for at least PER and maybe other MISC, it won't make much sense because the only supervision source will come from the spaCy model.
2. If I understand correctly, we then train a spaCy model with mainly the `hmm` labels, falling back to the pre-trained spaCy annotations if we have no ents for a record. So the only source of PER and other MISC is this fraction of the records (those with no annotation from the hmm) and as said earlier only from this one source of supervision.
3. Another general thing: NER training sets and models largely benefit from unlabelled examples, this is examples where there shouldn't be any entity. I'd like to make sure we are training the end models with examples with no entities.
4. More of a question: couldn't we do the MISC alignment before defining the labelling functions, instead of before training?

I guess that's it. Recap:

1. Streamline the pre-processing part.
2. Make use of Rubrix search capabilities from the start to define the heuristics (we can later in the same PR provide a video)
3. If we decide to evaluate on conll we should include heuristics for PER and MISC, and include them in the HMM aggregation process. Other option is focus only on improving the performance for the ORG label, and thus: 1) evaluate ORG performance with the pre-trained spaCy model (by using only that entity label from conll), 2) build an skweak training set with lfs and the pre-trained model, and 3) evaluate ORG performance with models trained on this dataset (with the hope they improve over the pre-trained model baseline).

We can go through this tomorrow.

**Comment by ruanchaves** [comment]
Hi @dvsrepo , thanks for the feedback. I will implement your suggestions and send a new PR.  I believe you correctly understood what I did in this tutorial. 

I will go with the option of focusing only on the ORG label. It seems to me that the process of figuring out the right lfs for the MISC, PER and LOC labels in CoNLL can become too time-consuming without adding much value to the tutorial.

> Question: In the gazetteers part, if the conclusion after exploring is that only the wiki gazetteer is working, we should explain what to do, do we don't make use of the other gazetteers?    

As there is no overlap between the entities in the other gazetteers and our training data, the final conclusion is that they are not useful for this particular dataset and should be ignored.    

I'm considering to simplify this section by including only the wiki gazetteer.

> More of a question: couldn't we do the MISC alignment before defining the labelling functions, instead of before training?    

Yes, sure. I don't think there is any impediment.
